components/num-util/src/lib.rs

```diff
@@ -56,5 +56,11 @@ pub enum Error {
     /// assert_eq!(Error::Limit, dec1.multiply_pow10(std::i16::MAX).unwrap_err());
     /// ```
     Limit,
+    /// The input of a string that is supposed to be converted to FixedDecimal is not accepted.
+    /// Any string with non-digit characters (except for one '.' and one '-' at the beginning of the string) is not accepted.
```

Comment:
I just realized that you should add a test with invalid syntax strings.  Make a parameterized test case with an array of several invalid strings and assert that you get Error::Syntax on each.